src/NBitcoin/ChainBase.cs

```diff
@@ -95,35 +131,22 @@ public bool SameTip(ChainBase chain)
             return this.Tip.HashBlock == chain.Tip.HashBlock;
         }
 
+        /// <summary>
+        /// Gets the required work for a network and a specific block height.
+        /// </summary>
+        /// <param name="network">The network to use for work computation.</param>
+        /// <param name="height">The height of the block to get work for.</param>
+        /// <returns>The required work.</returns>
         public Target GetWorkRequired(Network network, int height)
         {
             return this.GetBlock(height).GetWorkRequired(network);
         }
 
-        public bool Validate(Network network, bool fullChain = true)
-        {
-            ChainedBlock tip = this.Tip;
-            if (tip == null)
-                return false;
-
-            if (!fullChain)
-                return tip.Validate(network);
-
-            foreach (ChainedBlock block in tip.EnumerateToGenesis())
-            {
-                if (!block.Validate(network))
-                    return false;
-            }
-
-            return true;
-        }
-
-
         /// <summary>
-        /// Returns the first common block between two chains.
+        /// Returns the first common chained block header between two chains.
         /// </summary>
         /// <param name="chain">The other chain.</param>
-        /// <returns>First common block or <c>null</c>.</returns>
+        /// <returns>First common chained block header or <c>null</c>.</returns>
         public ChainedBlock FindFork(ChainBase chain)
```

Comment:
this method is implemented terribly, it goes through whole chain!? fortunately it is used only in tests - it should be moved to tests

please do it as a part of this PR because this method is terrible